ipld-importer/import.go

```diff
@@ -0,0 +1,89 @@
+package importer
+
+import (
+	"context"
+	"io"
+	"strings"
+
+	"github.com/ipfs/go-ipfs-cmdkit/files"
+	"github.com/ipfs/go-ipfs/core/coreunix"
+	ipld "github.com/ipfs/go-ipld-format"
+)
+
+// ToPrint is a first step towards a streaming importer.  Verify that we
+// can hijack the blockstore abstraction to redirect blocks as they arrive
+// Closely follows go-ipfs/core/commands/add.go: Run func
+func ToPrint(f files.File) error {
+	dserv := &pDAGService{
+		membership: make(map[string]bool),
+	}
+
+	ctx := context.Background() // using background for now, should upgrade later
```

Comment:
maybe de-duplicate the code in this function since it's the same in both

Reply:
With upcoming changes (inserted select statement, not returning errors from goroutine) deduplication looks like it is not quite worth the cost.  Also I am leaning towards your suggestion of removing the printing-dagservice and the ToPrint call later on in favor of debug logging in to Channel.